lib/google_crawler/search/search_keyword_task.ex

```diff
@@ -0,0 +1,18 @@
+defmodule GoogleCrawler.Search.SearchKeywordTask do
+  alias GoogleCrawler.Search.Keyword
+  alias GoogleCrawler.Google.Scrapper
+
+  def perform(%Keyword{} = keyword) do
+    case google_api_client().search(keyword.keyword) do
+      {:ok, body} ->
+        Scrapper.scrap(body)
+
+      {:error, reason} ->
+        raise "Keyword search failed: #{reason}"
+    end
+  end
+
+  defp google_api_client do
+    Application.get_env(:google_crawler, :google_api_client)
```

Comment:
I noticed that you used the same technic to inject the right module based on the environment (as per [this blog post I shared](https://github.com/olivierobert/crawlexir/pull/15/commits/369d8c77d5f6dc88df706b3e333dd6ac295e0977)). I even feel that this is something we could bring to Ruby projects

Reply:
Another interesting ideas -> https://medium.com/flatiron-labs/rolling-your-own-mock-server-for-testing-in-elixir-2cdb5ccdd1a0 🤔 This one is to running a server in the test env.